src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -86,6 +86,11 @@ namespace xrt {
     }
     args = xrt_core::xclbin::get_kernel_arguments((char *)buf,prop.size,sk_name);
     num_args = args.size();
+    if(args[num_args-1].type == xrt_core::xclbin::kernel_argument::argtype::global)
+      return_offset = (args[num_args-1].offset+PS_KERNEL_REG_OFFSET+16)/4;
+    else
+      return_offset = (args[num_args-1].offset+PS_KERNEL_REG_OFFSET+((args[num_args-1].size>4)?8:4))/4;
```

Comment:
Magic numbers.
Lots happening in this line. A comment for future developers would be helpful.